### ISSUE TITLE
Incorrect saturations for problems containing FOOL

### ISSUE THREAD

**Issue by ibnyusuf** [vprover/vampire#454]
Under certain circumstances, we incorrectly claim saturations for FOOL problems.

This happens when we use a preprocessing technique that gets rid of equalities such as inequality splitting or equality proxy. When setting up the saturation algorithm, we do not add superposition, as the problem no longer contains equalities. However, FOOLParamodulation can reintroduce equalities into the problem. Without superposition (and related rules) to deal with these equalities we can then saturate incorrectly.

**Comment by easychair** [comment]
What is a FOOL problem?

As a side note, inequality splitting preserves satisfiability. Equality
proxy can turn an unsatisfiable problem into a satisfiable one.

Andrei

**Comment by ibnyusuf** [comment]
A problem containing terms of [interpreted Boolean sort](https://arxiv.org/abs/1510.04821).

Right, so we don't need to worry about equlity proxy in that case since we can't saturate. Only inequality splitting.

@quickbeam123, this seems to be pretty much a corner case. Perhaps the solution that we implemented on the HOL branch will be suitable for master as well.

**Comment by quickbeam123** [comment]
I think, this still stays relevant with ep=RSTC. But yes, the fix from the HOL branch could be tried. It seems we are talking of a very cornerny corner case.

**Comment by easychair** [comment]
Yes, ep=RSTC preserves completeness, but it is hardly useful for anything.

A

**Comment by mezpusz** [comment]
I think this is partly solved by adding the corner cases [here](https://github.com/vprover/vampire/blob/e4ecdd2396288e7f9d91c0e8e748c6c82bbf56cb/Saturation/SaturationAlgorithm.hpp#L60) from #662 .
